generic_tracers/generic_tracer_utils.F90

```diff
@@ -3386,21 +3494,34 @@ subroutine g_tracer_vertdiff_G(g_tracer, h_old, ea, eb, dt, kg_m2_to_H, m_to_H,
 
     enddo; enddo ! i,j
 
-    !
-    !   Calculate the implicit vertical diffusion term
-    !   (Note: not sure if this needs any unit conversion)
-    !
+   !
+   !   Calculate the implicit vertical diffusion term
+   !   (Note: not sure if this needs any unit conversion)
+   !
 
-    if (do_diagnostic) then
+   if (g_tracer%diag_id_vdiffuse_impl .gt. 0 .or. g_tracer%diag_id_vdiffusec_impl .gt. 0) then
       do j = g_tracer_com%jsc, g_tracer_com%jec
          do i = g_tracer_com%isc, g_tracer_com%iec
             do k = 1, g_tracer_com%nk
-               g_tracer%vdiffuse_impl(i,j,k) = g_tracer_com%grid_tmask(i,j,k) *   &
-                    (g_tracer%field(i,j,k,tau) - g_tracer%vdiffuse_impl(i,j,k)) / dt
+               !liao: let h_old into vdiffuse and this term is content tendency
+               g_tracer%vdiffusec_impl(i,j,k) = g_tracer_com%grid_tmask(i,j,k) * &
+                    (g_tracer%field(i,j,k,tau) - g_tracer%vdiffuse_impl(i,j,k))/ dt
+               g_tracer%vdiffuse_impl(i,j,k) = h_old(i,j,k)*g_tracer_com%grid_tmask(i,j,k) *   &
+                    (g_tracer%field(i,j,k,tau) - g_tracer%vdiffuse_impl(i,j,k))/ dt
             enddo
          enddo
       enddo
-    endif
+   endif
+   if (g_tracer%diag_id_vdiffuseh_impl .gt. 0) then
+      do j = g_tracer_com%jsc, g_tracer_com%jec
+         do i = g_tracer_com%isc, g_tracer_com%iec
+            do k = 1, g_tracer_com%nk
+               g_tracer%vdiffuseh_impl(i,j,k) = h_old(i,j,k)*g_tracer_com%grid_tmask(i,j,k) *   &
```

Comment:
@andrew-c-ross,
Thank you! The new logic looks excellent. I have a quick question: I noticed that the calculation of `g_tracer%vdiffuseh_impl` seems identical to that of `g_tracer%vdiffuse_impl`:

https://github.com/NOAA-CEFI-Regional-Ocean-Modeling/ocean_BGC/blob/3c965b777c7b5b9f5620d712658527c302269bbe/generic_tracers/generic_tracer_utils.F90#L3509-L3510
I also confirmed this in my 1-D case. Am I perhaps overlooking something?

Reply:
Yeah the computation looks the same. The only difference I see is the units in the metadata (mol/kg m s-1 for vdiffuseh and mole/m^2/s for vdiffuse). I think mol/kg m s-1 should be right in Boussinesq mode.

Reply:
Thanks, @andrew-c-ross. I also noticed the unit difference in the metadata, and it seems like `vdiffuseh` is the one consistent with the computation under Boussinesq mode. My question is, should we keep just one of them for now, or should we keep both?

Reply:
We've been using `vdiffuseh` for the budget, but it would be easy to switch. Rather than adding a new diagnostic `vdiffuseh` how about just correcting the units for `vdiffuse`?

Reply:
Sounds good to me! It's more straightforward and requires fewer code changes.